File: radix_sort.py
#radix sort in python
def radix_sort(a):
    big=max(a)
    size=len(str(big))
    res_arr=[]
    
    for i in range(size):
        
        buckets=[[] for j in range(10)]
        
        for num in a:
            digit_value=(num//10**i)%10
            buckets[digit_value].append(num)
            
            
        # Concatenate the buckets in order to form the sorted array for this digit
        for bucket in buckets:
            res_arr.extend(bucket)
    return res_arr
    

a=list(map(int,input().split()))
res=radix_sort(a)
print(res)

File: radix_sort.java
#radix sort using java
import java.util.*;

public class RadixSort {
    public static ArrayList<Integer> radixSort(ArrayList<Integer> a) {
        int big = Collections.max(a);
        int size = String.valueOf(big).length();
        ArrayList<Integer> resArr = new ArrayList<>();
        
        for (int i = 0; i < size; i++) {
            ArrayList<ArrayList<Integer>> buckets = new ArrayList<>(10);
            for (int j = 0; j < 10; j++) {
                buckets.add(new ArrayList<>());
            }
            
            for (int num : a) {
                int digitValue = (int) ((num / Math.pow(10, i)) % 10);
                buckets.get(digitValue).add(num);
            }
            
            for (ArrayList<Integer> bucket : buckets) {
                resArr.addAll(bucket);
            }
        }
        return resArr;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        ArrayList<Integer> a = new ArrayList<>();
        while (scanner.hasNextInt()) {
            a.add(scanner.nextInt());
        }
        ArrayList<Integer> res = radixSort(a);
        System.out.println(res);
    }
}
